core/sched.c

```diff
@@ -218,6 +222,11 @@ thread_t *__attribute__((used)) sched_run(void)
             MPU_ATTR(1, AP_RO_RO, 0, 1, 0, 1, MPU_SIZE_32B) /* Attributes and Size */
             );
 #endif
+
```

Comment:
This is terribly in performant on the hifive1b. Stack guarding via PMP should only be used when context switches are infrequent.

Reply:
I suspect that the compiler doesn't inline the `write_pmpaddr` and `set_pmpcfg` functions, which in turn means that it probably can't optimize the switch-case statement in that function when a fixed pmp slot number is provided.